cmd/restic/cmd_mount.go

```diff
@@ -0,0 +1,301 @@
+	knownSnapshots map[string]*restic.Snapshot
+}
+
+func (sn *snapshots) Attr(ctx context.Context, a *fuse.Attr) error {
+	a.Inode = 0
+	a.Mode = os.ModeDir | 0555
+	return nil
+}
+
+func (sn *snapshots) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
+	ret := make([]fuse.Dirent, 0)
+	for id := range sn.repo.List(backend.Snapshot, ctx.Done()) {
+		snapshot, err := restic.LoadSnapshot(sn.repo, id)
+		if err != nil {
+			return nil, err
+		}
+		sn.knownSnapshots[snapshot.Time.Format(time.RFC3339)] = snapshot
+		ret = append(ret, fuse.Dirent{
+			Inode: binary.BigEndian.Uint64(id[:8]),
+			Type:  fuse.DT_Dir,
+			Name:  snapshot.Time.Format(time.RFC3339),
+		})
+	}
+
+	return ret, nil
+}
+
+func (sn *snapshots) Lookup(ctx context.Context, name string) (fs.Node, error) {
+	if _, ok := sn.knownSnapshots[name]; !ok {
+		// At least this snapshot is not cached. We use this opportunity to
+		// load all missing snapshots
+		for id := range sn.repo.List(backend.Snapshot, ctx.Done()) {
+			snapshot, err := restic.LoadSnapshot(sn.repo, id)
+			if err != nil {
+				return nil, err
+			}
+			snapshotName := snapshot.Time.Format(time.RFC3339)
+			sn.knownSnapshots[snapshotName] = snapshot
+			break
+		}
+	}
+
+	snapshot := sn.knownSnapshots[name]
+	tree, err := restic.LoadTree(sn.repo, snapshot.Tree)
+	if err != nil {
+		return nil, err
+	}
+	return &dir{
+		repo: sn.repo,
+		tree: tree,
+	}, nil
+	return nil, fuse.ENOENT
+}
+
+// Statically ensure that *dir implement those interface
+var _ = fs.HandleReadDirAller(&dir{})
+var _ = fs.NodeStringLookuper(&dir{})
+
+type dir struct {
+	repo  *repository.Repository
+	tree  *restic.Tree
+	inode uint64
+}
+
+func (d *dir) Attr(ctx context.Context, a *fuse.Attr) error {
+	a.Inode = d.inode
+	a.Mode = os.ModeDir | 0555
+	return nil
+}
+
+func (d *dir) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
+	ret := make([]fuse.Dirent, 0, len(d.tree.Nodes))
+
+	for _, node := range d.tree.Nodes {
+		var typ fuse.DirentType
+		switch {
+		case node.Mode.IsDir():
+			typ = fuse.DT_Dir
+		case node.Mode.IsRegular():
+			typ = fuse.DT_File
+		}
+
+		ret = append(ret, fuse.Dirent{
+			Inode: node.Inode,
+			Type:  typ,
+			Name:  node.Name,
+		})
+	}
+
+	return ret, nil
+}
+
+func (d *dir) Lookup(ctx context.Context, name string) (fs.Node, error) {
+	for _, node := range d.tree.Nodes {
+		if node.Name == name {
+			switch {
+			case node.Mode.IsDir():
+				subtree, err := restic.LoadTree(d.repo, node.Subtree)
+				if err != nil {
+					return nil, err
+				}
+				return &dir{
+					repo:  d.repo,
+					tree:  subtree,
+					inode: binary.BigEndian.Uint64(node.Subtree[:8]),
+				}, nil
+			case node.Mode.IsRegular():
+				return makeFile(d.repo, node)
+			}
+		}
+	}
+
+	return nil, fuse.ENOENT
+}
+
+// Statically ensure that *file implements the given interface
+var _ = fs.HandleReader(&file{})
+
+type file struct {
+	repo *repository.Repository
+	node *restic.Node
+
+	sizes []uint32
+
+	// cleartext contents
```

Comment:
instead of this comment, why not rename the variable to `cleartextContent` and make it self-explanatory?